### DISCUSSION TITLE
Problem with @tailwind and @layer directives when using in different files

### DISCUSSION THREAD

**Discussion by existe-deja** [tailwindlabs/tailwindcss#6101] [Help] [answered]
Hi,

I was wondering if I had to add `@tailwind components` directive each time I want to use the `@component layer` directive?

I need to split my files into several ones and if I ommit to re-declare @tailwind directive it will fail with error `@layer components is used but no matching @tailwind components directive is present.`.

I'm using `@nuxtjs/tailwindcss` package.

Thanks for the help

**Comment by riabiy-denis** [comment, marked as answer]
Was scratching my head too + @adamwathan 's answer didn't help (he answered in a separate issue, can't find it). 
This is the best guide
https://bloggie.io/@kinopyo/organize-your-css-in-the-tailwind-style-with-layer-directive

tl;dr

```
@tailwind base;
@tailwind components;
@tailwind utilities;

// Base
@layer base {
  @import "base/variables";
  @import "stuff/you/want/go/here/kinda/organize/in/this/file/css/in/others/if/needed";
}

```

**Reply by sina-salahshour** [reply]
is there anyway to make this work with css modules?
